File: eimui/src/main/java/com/xjl/eimui/util/GlideHelper.java
package com.xjl.eimui.util;

import com.bumptech.glide.request.RequestOptions;
import com.xjl.eimui.R;

import java.util.HashMap;

public enum GlideHelper {

    INSTANCE;

    private RequestOptions onlyCenterCrop = null;

    public RequestOptions getOnCenterCrop() {
        if (onlyCenterCrop == null) {
            onlyCenterCrop = new RequestOptions();
            this.onlyCenterCrop.skipMemoryCache(false);
            this.onlyCenterCrop.centerCrop();
        }

        return onlyCenterCrop;
    }

    HashMap<Integer, RequestOptions> hashMap = new HashMap<>();

    public RequestOptions getErrorOptions(int res) {
        if (hashMap.get(res) == null) {
            RequestOptions requestOptions = new RequestOptions();
            requestOptions.skipMemoryCache(false);
            requestOptions.error(R.mipmap.avatar_spiderman);
            hashMap.put(res, requestOptions);
        }
        return hashMap.get(res);
    }


}

File: eimui/src/main/java/com/xjl/eimui/util/GlideHelper.kt
package com.xjl.eimui.util

import com.bumptech.glide.request.RequestOptions
import com.xjl.eimui.R

enum class GlideHelper {
    INSTANCE;

    private var onlyCenterCrop: RequestOptions? = null
    val onCenterCrop: RequestOptions
        get() {
            if (onlyCenterCrop == null) {
                onlyCenterCrop = RequestOptions().apply{
                    skipMemoryCache(false)
                    centerCrop()
                }
            }
            return onlyCenterCrop!!
        }
    var hashMap = HashMap<Int, RequestOptions?>()
    fun getErrorOptions(res: Int): RequestOptions? {
        if (hashMap[res] == null) {
            val requestOptions = RequestOptions()
            requestOptions.skipMemoryCache(false)
            requestOptions.error(R.mipmap.avatar_spiderman)
            hashMap[res] = requestOptions
        }
        return hashMap[res]
    }
}
